application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,262 @@
+            event.reply("You dont have the right permissions to use this command")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        Member selfMember = Objects.requireNonNull(event.getGuild()).getSelfMember();
+        if (!selfMember.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("The bot needs the manage role permissions").setEphemeral(true).queue();
+            logger.warn("The bot needs the manage role permissions");
+            return;
+        }
+
+        SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+        boolean ephemeral = false;
+
+        if (Objects.equals(event.getSubcommandName(), CHOOSE_OPTION)) {
+            addMenuOptions(event, menu, "Select the roles to display", 1);
+            ephemeral = true;
+        } else {
+            addMenuOptions(event, menu, "Select your roles", 0);
+        }
+
+        // Handle Optional arguments
+        OptionMapping titleOption = event.getOption(TITLE_OPTION);
+        OptionMapping descriptionOption = event.getOption(DESCRIPTION_OPTION);
+
+        String title = handleOption(titleOption);
+        String description = handleOption(descriptionOption);
+
+        if (ephemeral) {
+            event.replyEmbeds(makeEmbed(title, description))
+                .addActionRow(menu.build())
+                .setEphemeral(true)
+                .queue();
+        } else {
+            event.getChannel()
+                .sendMessageEmbeds(makeEmbed(title, description))
+                .setActionRow(menu.build())
+                .queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+        }
+    }
+
+    /**
+     * Adds role options to a selection menu
+     * <p>
+     *
+     * @param event the {@link SlashCommandEvent}
+     * @param menu the menu to add options to {@link SelectionMenu.Builder}
+     * @param placeHolder the placeholder for the menu {@link String}
+     * @param minValues the minimum number of selections. nullable {@link Integer}
+     */
+    private static void addMenuOptions(@NotNull SlashCommandEvent event,
+            @NotNull SelectionMenu.Builder menu, @NotNull String placeHolder,
+            @Nullable Integer minValues) {
+
+        Role highestBotRole =
+                Objects.requireNonNull(event.getGuild()).getSelfMember().getRoles().get(1);
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
+
+        List<Role> roles = new ArrayList<>(
+                guildRoles.subList(guildRoles.indexOf(highestBotRole) + 1, guildRoles.size()));
+
+        if (minValues != null) {
+            menu.setMinValues(minValues);
+        }
+
+        menu.setPlaceholder(placeHolder).setMaxValues(roles.size());
+
+        for (Role role : roles) {
+            if (role.getName().equals("@everyone") || role.getTags().getBotId() != null) {
```

Comment:
```suggestion
            if (role.getIdLong() == event.getGuild().getIdLong() || role.getTags().getBotId() != null) {
```
You can easily create a role named `@everyone`
So instead, check or the role's ID is equal to the guild's ID.

Thankfully, the real `@everyone` role is always equal to the guild's ID


And to use Role#getTags you need to enable this in the cache, which I highly doubt is enabled.